src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -929,7 +931,8 @@ async Task<(bool, ImmutableArray<ProjectConfiguration>)> CheckAsync(UpToDateChec
                 {
                     logger.Verbose(nameof(Resources.FUTD_Completed), sw.Elapsed.TotalMilliseconds);
 
-                    _lastFailureReason = logger.FailureReason ?? "";
+                    _lastFailureReason = logger.FailureReason;
+                    _lastFailureDescription = logger.FailureDescription;
```

Comment:
What happens when failure reason or description are null?

Reply:
There's `Assumes.NotNull` on both at a later point to validate the nullability of these values.